tests/test_solver.py

```diff
@@ -31,17 +32,58 @@ def heat_balance(conductor_temperature, current):
     assert conductor_temperature == pytest.approx(9000, rel=1e-7)
 
 
-def test_compute_conductor_temperature_caps_ampacity_at_zero():
-    def heat_balance(conductor_temperature, current):
+def test_bisect_raises_value_error():
+    def heat_balance(current):
         A = current
-        T = conductor_temperature
-        return (A + 100 * T) * (current + 200 * T)
+        T = 90
+        return (A + 100 * T) * (A + 100 * T)
 
-    conductor_temperature = solver.compute_conductor_ampacity(
+    with pytest.raises(ValueError):
+        solver.bisect(
+            heat_balance,
+            xmin=0,
+            xmax=10_000,
+            tolerance=1e-8,
+        )
+
+
+def test_bisect_handles_function_returning_array_happy_path():
+    def heat_balance(currents: np.array):
+        A = currents
+        T = 90
+        res = (A - 100 * T) * (currents + 100 * T)
+        return res
+
+    solution = solver.bisect(
         heat_balance,
-        max_conductor_temperature=90,
-        min_ampacity=0,
-        max_ampacity=10_000,
+        xmin=np.array([0, 0]),
+        xmax=np.array([10_000, 10_000]),
         tolerance=1e-8,
     )
-    assert conductor_temperature == pytest.approx(0, rel=1e-7)
+    np.testing.assert_array_almost_equal(solution, [9000, 9000], decimal=8)
```

Comment:
Just a little nitpicking for consistency with the other numbers which are more readable:
```suggestion
    np.testing.assert_array_almost_equal(solution, [9_000, 9_000], decimal=8)
```